backend/src/vscode-youi-events.ts

```diff
@@ -27,6 +41,36 @@ export class VSCodeYouiEvents implements YouiEvents {
         this.showInstallMessage();
     }
 
+    public showProgress(message?: string): void {
+        if (this.isOutputShown) {
+            this.logger.trace("Output was already shown.");
+            // Not to show more then one time Generation Notification
+            return;
+        }
+        const openOutput: any = this.messages.show_progress_button;
+        const buttons: string[] = [];
+        buttons.push(openOutput);
+        if (_.isEmpty(message)) {
+            message = this.messages.show_progress_message;
+        }
+        this.logger.debug("Showing Progress.", {
+            notificationMessage: message,
+        });
+        vscode.window.showInformationMessage(message, ...buttons).then(selection => {
+            if (selection === openOutput) {
+                return this.toggleOutput();
+            }
+        });
+    }
+
+    private toggleOutput() {
+        this.isOutputShown = true;
+        console.log('this.isOutputShown');
+        console.log(this.isOutputShown);
```

Comment:
this logic should be defined in vscode-output.ts because we do output.show() from several modules

Reply:
From Avital:
We agreed to allow the extension owner to use the notification that opens the output at any time.
Since we don’t know if the output is opened, we’ll display the notification when the extension owner asked to show it.